### PR TITLE
[DS-4239] Migrate the workflow.xml to spring

### PR DESCRIPTION
**Pull request by KevinVdV**
Moving the workflow.xml file from a custom xml configuration to spring.

JIRA: https://jira.duraspace.org/browse/DS-4239

**TODO**

- [X] Code changes
- [x] Migration xsl file
- [ ] Documentation update

### CODE REVIEW

**Comment by KevinVdV** [comment]
Still need to do some additional testing & creating the migration XSL file, but feedback on the code is certainly welcome.

**Comment by KevinVdV** [comment]
The migration file to migrate a custom workflow.xml to the spring based version of the file has been attached to the JIRA ticket (credit to @MarieVerdonck who helped me in creating this file).

**Review by benbosman — COMMENTED** [review]
Thanks @KevinVdV for this contribution.

I started with a code review and included a question inline.

Have you verified that actually using the workflow still works will when one of the outcomes of a steps is to go back to a previous step (e.g. the selectSingleReviewer bean)

My next part of the review will be to try and use it (but this is of course rather complicated since Angular support of the workflow is not completed)

File: dspace-api/src/test/data/dspaceFolder/config/spring/api/workflow.xml [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+                <ref bean="editstep"/>
+                <ref bean="finaleditstep"/>
+            </util:list>
+        </property>
+    </bean>
+
+    <bean name="reviewstep" class="org.dspace.xmlworkflow.state.Step">
+        <property name="userSelectionMethod" ref="claimaction"/>
+        <property name="role">
+            <bean name="reviewer" class="org.dspace.xmlworkflow.Role">
```
[the remark this answers is not in the document]
Comment (KevinVdV) [reply dspace-api/src/test/data/dspaceFolder/config/spring/api/workflow.xml]: Done

**Comment by KevinVdV** [comment]
When testing the selective review I found a couple of issues, I resolved these in the 2 commits that were added, so this PR is now ready for another review.

**Review by benbosman — APPROVED** [review]
I have tested the functionality in detail (as much as possible with the limited UI support for the workflow).
I encountered problems testing this with an existing database because flyway didn't handle the old SQL file correctly. This is something to keep in mind when testing the functionality, but it's not a bug in this PR.

This looks good to me, thanks for improving the backend

**Review by tdonohue — APPROVED** [review]
👍 Code looks good to me. As @benbosman noted this is difficult to test as the REST API is lacking in this area at this time.

After merging this, we need to ensure all DSpace 7 developers either: Recreate their test database OR run `./dspace database migrate ignored`.  As Ben notes, there's a database migration in this PR which is not triggered by default (as it has an older date than the most recent migration).  However, it will be triggered just fine in a fresh install or fresh upgrade.

Documentation also is required after this PR is merged, as this change in configuration should be documented in the upgrade procedure (minimally). https://wiki.lyrasis.org/display/DSDOC7x/Upgrading+DSpace

**Pull request merged**